### PR TITLE
[BCHDCC-111] Distance filter not working for 0.5 miles

### PR DESCRIPTION
**Pull request by micaelacunhadev**
## Description

 - Fixed distance filter when a value of 0.5 miles is selected
 - Improved accessibility on search filters form

## Motivation and Context

[BCHDCC-111](https://app.clickup.com/t/9006094761/BCHDCC-111) - Distance filter not working for 0.5 miles

## Screenshots

**Libraries Added**
None

## Tests to Run
None

## Internal Checklist

- [ ] My code follows the code style of this project (https://rubystyle.guide/).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/assets/stylesheets/components/_input-search.scss [hunk 1]
```diff
@@ -332,10 +332,6 @@ input.search-input:-webkit-autofill {
 }
 
 .input-search-filter.input-search-filter-option {
-  > p {
-    margin: -12px 0 12px 0;
-  }
-
```
Comment (micaelacunhadev) [inline app/assets/stylesheets/components/_input-search.scss:338]: This was used for the title of the search filters fieldset, but it is no longer needed now that the fieldset title was moved into a `legend` element.

File: app/searches/locations_search.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ class LocationsSearch
   attribute :accessibility, type: Array
   attribute :lat, type: Float
   attribute :long, type: Float
-  attribute :distance, type: Integer
+  attribute :distance, type: Float
```
Comment (micaelacunhadev) [inline app/searches/locations_search.rb:18]: Fixes the radius filter!

File: app/views/component/search/_category.html.haml [hunk 3]
```diff
@@ -1,25 +1,34 @@
 = field_set_tag nil, id: 'category-options', class: 'input-search-filter input-search-filter-option' do
-  %p.filter-results-label
-    Filter Results
-  %legend.sr-only= t('labels.category')
+  %legend.filter-results-label= t('labels.filters.filters_menu_title')
```
Comment (micaelacunhadev) [inline app/views/component/search/_category.html.haml:2]: For better accessibility, a fieldset should have a legend, so I changed "Filter Results" to a `legend` element. Also added the text for it to `en.yml`.

File: app/views/component/search/_category.html.haml [hunk 4]
```diff
@@ -1,25 +1,34 @@
...
-          error: function(xhr, status, error) {
-            console.error("Error fetching subcategories:", error);
+          error: function(jqXHR, textStatus, errorThrown) {
+            if (jqXHR.status === 422) {
+              // Handle 422 error
+              console.error('Unprocessable Entity:', jqXHR.responseJSON);
+            } else {
+              // Handle other errors
+              console.error(textStatus, errorThrown);
+            }
```
Comment (micaelacunhadev) [inline app/views/component/search/_category.html.haml:36]: There were errors from this request, even if in the front end everything was working, so I added headers and some more details about the errors to debug it.

File: app/views/component/search/_category.html.haml [hunk 5]
```diff
@@ -30,20 +39,21 @@
     function updateSubcategories(subcategories) {
       var $subcategoriesContainer = $('#subcategoriesList');
       $subcategoriesContainer.empty();
+      var selec
```
Comment (micaelacunhadev) [inline app/views/component/search/_category.html.haml]: Left here by mistake - removed in next commit.

File: app/views/component/search/_category.html.haml [hunk 6]
```diff
@@ -30,20 +39,21 @@
...
 
       if (subcategories && subcategories.length > 0) {
-        subcategories.forEach(function(subcat) {
+        subcategories.forEach( (subcat, i) => {
           var $div = $('<div>', { class: 'filter-category-item' });
-          var $subcatName = subcat.replace(/\s+/g, "_")
```
Comment (micaelacunhadev) [inline app/views/component/search/_category.html.haml]: Using the actual category names was causing issues with matching the subcategories ids to the respective `input`, so I removed this line and now the ids are based on the subcategory index position in the list of subcategories.

File: app/views/component/search/_category.html.haml [hunk 7]
```diff
@@ -30,20 +39,21 @@
...
+          var $selectedCategories = #{raw @selected_categories}
```
Comment (micaelacunhadev) [inline app/views/component/search/_category.html.haml]: Without `raw` here, the Ruby array is not properly read and it was causing errors. Surprisingly, the code would still work in the frontend, but this makes the code more resilient.

File: app/views/component/search/_geolocate.html.haml [hunk 8]
```diff
@@ -1,27 +1,29 @@
...
-    - else 
+    - else
       %svg{width: "26", height: "26", viewBox: "0 0 22 22", fill: "none", xmlns: "http://www.w3.org/2000/svg", id: "button-geolocate", "aria-label" => t('buttons.services_near_me'), title: t('buttons.services_near_me')}
         %path{d: "M19.94 10C19.48 5.83 16.17 2.52 12 2.06V0H10V2.06C5.83 2.52 2.52 5.83 2.06 10H0V12H2.06C2.52 16.17 5.83 19.48 10 19.94V22H12V19.94C16.17 19.48 19.48 16.17 19.94 12H22V10H19.94ZM11 18C7.13 18 4 14.87 4 11C4 7.13 7.13 4 11 4C14.87 4 18 7.13 18 11C18 14.87 14.87 18 11 18Z", fill: "#222222"}
 
-    = text_field_tag :address, @address, class: "search-address-input"
-  = label_tag nil, t('labels.search_hint_location'), class: 'search-hint'
+    = text_field_tag :address, @address, class: "search-address-input",  autocomplete: "off"
+  %span.search-hint
+    = t('labels.search_hint_location')
```
Comment (micaelacunhadev) [inline app/views/component/search/_geolocate.html.haml:17]: Replaced duplicate `label` with `span`.

File: app/views/component/search/_geolocate.html.haml [hunk 9]
```diff
@@ -1,27 +1,29 @@
...
+  = select_tag :distance,
                options_for_select([["0.5 miles", 0.5], ["1 mile", 1], ["2 miles", 2], ["5 miles", 5], ["10 miles", 10], ["15 miles", 15], ["20 miles", 20], ["25 miles", 25], ["30 miles", 30]], @selected_distance_filter), 
-               include_blank: 'All', 
+               include_blank: 'All',
                class: "category-search-select-left-menu",
                aria: { describedby: 'Distance from location selected' }
 
-  = label_tag nil, t('labels.search_hint_distance'), class: 'search-hint'               
+  %span.search-hint
+    =t('labels.search_hint_distance')
```
Comment (micaelacunhadev) [inline app/views/component/search/_geolocate.html.haml:29]: Replaced duplicate `label` with `span`.

File: app/views/component/search/_keyword.html.haml [hunk 10]
```diff
@@ -1,6 +1,5 @@
 = field_set_tag nil, id: 'keyword-search-box', class: 'input-search-filter input-search-filter-option' do
-  %legend.sr-only= t('labels.search_services')
-  = label_tag 'keyword', t('labels.search_services'), class: 'filter-results-label'
+  %legend.filter-results-label= t('labels.search_services')
```
Comment (micaelacunhadev) [inline app/views/component/search/_keyword.html.haml:2]: Replaced `label` with `legend` for `fieldset`.

File: app/views/component/search/_filters.html.haml [hunk 11]
```diff
@@ -1,22 +1,21 @@
 %div#categoryFiltersContainerDiv.category-filters-container
 
 - if @main_category_selected_name == ""
-  %div#parent-category.parent-category-label-container{ tabindex: "0", role: "button", aria: { label: 'Select a Topic from above to display additional filters.', expanded: 'false'}}
-    %legend
-      %span#subcategoriesFilterTitle
+  %div#parent-category.parent-category-label-container{ tabindex: "1", aria: { label: 'Select a Topic from above to display additional filters.', expanded: 'false'}}
+    %span#subcategoriesFilterTitle
```
Comment (micaelacunhadev) [inline app/views/component/search/_filters.html.haml]: Revisiting this in next commit.

**Review by micaelacunhadev — COMMENTED** [review]
An overview of my changes.

File: app/views/component/search/_category.html.haml [hunk 12]
```diff
@@ -1,25 +1,34 @@
...
           data: { category_name: categoryName },
           dataType: 'json',
           success: function(data) {
             updateSubcategories(data.sub_cat_array);
             checkMatchedCategory();
           },
-          error: function(xhr, status, error) {
-            console.error("Error fetching subcategories:", error);
+          error: function(jqXHR, textStatus, errorThrown) {
+            if (jqXHR.status === 422) {
```
Comment (anna-dorigo) [inline app/views/component/search/_category.html.haml:25]: ✨

File: app/views/component/search/_geolocate.html.haml [hunk 13]
```diff
@@ -1,27 +1,29 @@
 = field_set_tag nil, id: 'geolocation-filter', class: 'input-search-filter input-search-filter-option' do
   %legend.sr-only= 'Geolocation Filter'
-  = label_tag 'address', 'Location', id: 'address-label'
+  = label_tag 'address', t('filters.location'), id: 'address-label'
```
Comment (anna-dorigo) [inline app/views/component/search/_geolocate.html.haml:3]: ✨

File: app/views/component/search/_geolocate.html.haml [hunk 14]
```diff
@@ -1,27 +1,29 @@
...
-  = label_tag nil, t('labels.search_hint_location'), class: 'search-hint'
+    = text_field_tag :address, @address, class: "search-address-input",  autocomplete: "off"
+  %span.search-hint
+    = t('labels.search_hint_location')
 
 = field_set_tag nil, id: 'distance-filter', class: 'input-search-filter input-search-filter-option' do
   %legend.sr-only= t('labels.distance_filter')
-  = label_tag 'distance', 'Radius', for: 'distance'
-  = select_tag :distance, 
+  = label_tag 'distance', t('filters.radius'), for: 'distance'
```
Comment (anna-dorigo) [inline app/views/component/search/_geolocate.html.haml:21]: ✨

File: app/views/component/search/_category.html.haml (hunk 4 above)
Comment (anna-dorigo) [reply app/views/component/search/_category.html.haml:36]: Can you please addd some example of the error being logged?

**Review by anna-dorigo — APPROVED** [review]
👍 Tested localy and it works great. 

Just a quick question that is not a blocker

File: app/views/component/search/_category.html.haml (hunk 4 above)
Comment (micaelacunhadev) [reply app/views/component/search/_category.html.haml:36]: @anna-dorigo can you please double-check this is what you were asking for?

**Pull request merged**